File: src/main/java/com/github/holly/accountability/user/AccountabilitySessionUser.java
package com.github.holly.accountability.user;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.List;


public class AccountabilitySessionUser implements UserDetails {

    private final Long id;
    private final String name;
    private final String username;
    private final String password;
    private final List<GrantedAuthority> authorities;

    public AccountabilitySessionUser(Long id, String name, String username, String password, List<GrantedAuthority> authorities) {
        this.id = id;
        this.name = name;
        this.username = username;
        this.password = password;
        this.authorities = authorities;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return this.authorities;
    }

    @Override
    public String getPassword() {
        return this.password;
    }

    @Override
    public String getUsername() {
        return this.username;
    }

    public String getName() {
        return name;
    }

    public Long getId() {
        return id;
    }
}

File: src/main/kotlin/com/github/holly/accountability/config/user/AccountabilitySessionUser.kt
package com.github.holly.accountability.config.user

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

data class AccountabilitySessionUser(
    val id: Long,
    val name: String,
    val _username: String,
    val _password: String,
    val _authorities: List<GrantedAuthority>
): UserDetails {

    override fun getAuthorities(): Collection<GrantedAuthority?>? {
        return _authorities
    }

    override fun getPassword(): String? {
        return _password
    }

    override fun getUsername(): String? {
        return _username
    }
}
